lib/nostrum/api.ex

```diff
@@ -1813,4 +1846,15 @@ defmodule Nostrum.Api do
     Application.get_env(:nostrum, :token)
   end
 
+  defp handle_request_with_decode(response, type) 
```

Comment:
I think this function name is a little verbose. For now it's fine as this is a WIP and something like `handle` is already in use. In the future we should maybe consider changing the name. 🙃

Reply:
I think verbose is okay for self-documentation purposes, but I am okay with a rename in the future. Renaming a function like this will not be hard.